exercises/t_jubb/ex02_oo_basics/src/main.cpp

```diff
@@ -0,0 +1,270 @@
+
+#include <algorithm>
+#include <cmath>
+#include <iostream>
+#include <string>
+#include <vector>
+
+
+/*
+Shape class
+ */
+class Shape {
+
+public:
+	Shape() { area = 0.0; this->perimeter = 0.0; this->sides = 0; this->type = " "; };
+
+	double get_area() { return area; };
+	double get_perimeter() { return perimeter; };
+	double get_sides() { return sides; };
+	std::string get_type() { return type; };
+	void print_shape() {  
+		int pad = 10 - type.size();
+		std::string pad_blank(pad, ' ');
+		std::cout << "Shape : " << type << pad_blank;
+		std::cout << " Area : " << area << " Perim. : " << perimeter << std::endl;
+		};
+
+protected:
+	double area, perimeter;
```

Comment:
It's generally considered bad practice to mark member data as protected as it doesn't really encapsulate it properly. In this specific case, as discussed above, the class doesn't need an implementation, but if it did each class should ideally keep all of its data `private`.